### PR TITLE
Update JSON API - contract template identifiers info

### PR DESCRIPTION
**Pull request by hrischuk-da**

### CODE REVIEW

File: docs/2.10.0/docs/json-api/index.rst [hunk 1]
```diff
@@ -506,7 +506,21 @@ HTTP Request
 
 Where:
 
-- ``templateId`` is the contract template identifier, which is formatted as ``"<package ID>:<module>:<entity>"``. As a convenience for interactive API exploration (such as with curl and similar tools), you can also omit the package ID (i.e. specifying the ``templateId`` as ``"<module>:<entity>"``) **if there is only one template with that name across all loaded packages**. Code should always specify the package ID, since it's common to have more versions of a template sharing the same module and entity name but with different package IDs. If the package identifier is not specified and the template cannot be uniquely identified without it, the HTTP JSON API service will report that the specified template cannot be found. **Omitting the package ID is not supported for production use.**
+- ``templateId`` is the contract template identifier which has the following
+  two formats: ``#<package name>:<module>:<entity>`` or ``<package
+  ID>:<module>:<entity>``. 
+  
+  The package name format works with the smart contract upgrading feature so
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:513]:
```suggestion
  The package name format supports the smart contract upgrading feature,
```

File: docs/2.10.0/docs/json-api/index.rst [hunk 2]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  that contracts with different package IDs but the same package name can be
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:514]:
```suggestion
  allowing contracts with different package IDs but the same package name to be
```

File: docs/2.10.0/docs/json-api/index.rst [hunk 3]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  used in a uniform manner.
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:515]:
```suggestion
  used uniformly. 
```

File: docs/2.10.0/docs/json-api/index.rst [hunk 4]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  two formats: ``#<package name>:<module>:<entity>`` or ``<package
+  ID>:<module>:<entity>``. 
+  
+  The package name format works with the smart contract upgrading feature so
+  that contracts with different package IDs but the same package name can be
+  used in a uniform manner. 
+  
+  Using the package ID format refers to a single package. 
+  
+  Previously JSON API had supported partially qualified template IDs, (i.e.
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:519]:
```suggestion
  Previously JSON API had supported partially qualified template IDs, such as
```

File: docs/2.10.0/docs/json-api/index.rst [hunk 5]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  simply ``<module>:<entity>``) as an interactive convenience which fails if
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:520]:
```suggestion
  ``<module>:<entity>``, as an interactive convenience, which fails if
```

File: docs/2.10.0/docs/json-api/index.rst [hunk 6]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  there is more than one package with matching template names. **Since this
+  format was not supported for production use and will not work with smart
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:522]:
```suggestion
  format is not supported for production use and does not work with smart
```
This suggestion follows the style guide and avoids using the future tense. If it significantly changes the meaning, please disregard it.

File: docs/2.10.0/docs/json-api/index.rst [hunk 7]
```diff
@@ -1056,8 +1070,12 @@ HTTP Request
 
 Where:
 
-- ``templateIds`` -- either an array of contract template identifiers or an array containing a single interface identifier to search through. Mixing of template ID's and interface ID's, or specifying more than one interface ID is not allowed.
-- ``query`` -- search criteria to apply to the specified ``templateIds``, formatted according to the :doc:`search-query-language`.
+- ``templateIds`` -- either an array of contract template identifiers in the
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:1073]:
```suggestion
- ``templateIds``: either an array of contract template identifiers in the
```

File: docs/2.10.0/docs/json-api/index.rst [hunk 8]
```diff
@@ -1056,8 +1070,12 @@ HTTP Request
...
+  same format as in the :ref:`create request <create-request>`.  It can also
+  be an array containing a single interface identifier to search through.
+  Mixing of template ID's and interface ID's, or specifying more than one
+  interface ID is not allowed.
+  - ``query`` -- search criteria to apply to the specified ``templateIds``, formatted according to the :doc:`search-query-language`.
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:1078]:
```suggestion
  - ``query``: search criteria to apply to the specified ``templateIds``, formatted according to the :doc:`search-query-language`.
```

File: docs/2.10.0/docs/json-api/index.rst [hunk 9]
```diff
@@ -1056,8 +1070,12 @@ HTTP Request
 
 Where:
 
-- ``templateIds`` -- either an array of contract template identifiers or an array containing a single interface identifier to search through. Mixing of template ID's and interface ID's, or specifying more than one interface ID is not allowed.
-- ``query`` -- search criteria to apply to the specified ``templateIds``, formatted according to the :doc:`search-query-language`.
+- ``templateIds`` -- either an array of contract template identifiers in the
+  same format as in the :ref:`create request <create-request>`.  It can also
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:1074]:
```suggestion
  same format as in the :ref:`create request <create-request>` or
```

File: docs/2.10.0/docs/json-api/index.rst [hunk 10]
```diff
@@ -1056,8 +1070,12 @@ HTTP Request
...
+  be an array containing a single interface identifier to search through.
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:1075]:
```suggestion
  an array containing a single interface identifier to search through.
```

File: docs/2.10.0/docs/json-api/index.rst [hunk 11]
```diff
@@ -1056,8 +1070,12 @@ HTTP Request
...
+  Mixing of template ID's and interface ID's, or specifying more than one
+  interface ID is not allowed.
+  - ``query`` -- search criteria to apply to the specified ``templateIds``, formatted according to the :doc:`search-query-language`.
 - ``readers`` -- *optional* non-empty list of parties to query as; must be a subset of the actAs/readAs parties in the JWT
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/json-api/index.rst:1079]:
```suggestion
- ``readers``: *optional* non-empty list of parties to query as; must be a subset of the actAs and readAs parties in the JWT.
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 12]
```diff
@@ -506,7 +506,21 @@ HTTP Request
 
 Where:
 
-- ``templateId`` is the contract template identifier, which is formatted as ``"<package ID>:<module>:<entity>"``. As a convenience for interactive API exploration (such as with curl and similar tools), you can also omit the package ID (i.e. specifying the ``templateId`` as ``"<module>:<entity>"``) **if there is only one template with that name across all loaded packages**. Code should always specify the package ID, since it's common to have more versions of a template sharing the same module and entity name but with different package IDs. If the package identifier is not specified and the template cannot be uniquely identified without it, the HTTP JSON API service will report that the specified template cannot be found. **Omitting the package ID is not supported for production use.**
+- ``templateId`` is the contract template identifier which has the following
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:509]:
```suggestion
- ``templateId`` is the contract template identifier, which has the following
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 13]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  two formats: ``#<package name>:<module>:<entity>`` or ``<package
+  ID>:<module>:<entity>``. 
+  
+  The package name format works with the smart contract upgrading feature so
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:513]:
```suggestion
  The package name format supports the smart contract upgrading feature,
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 14]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  that contracts with different package IDs but the same package name can be
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:514]:
```suggestion
  allowing contracts with different package IDs but the same package name to be
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 15]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  used in a uniform manner.
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:515]:
```suggestion
  used uniformly. 
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 16]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  two formats: ``#<package name>:<module>:<entity>`` or ``<package
+  ID>:<module>:<entity>``. 
+  
+  The package name format works with the smart contract upgrading feature so
+  that contracts with different package IDs but the same package name can be
+  used in a uniform manner. 
+  
+  Using the package ID format refers to a single package. 
+  
+  Previously JSON API had supported partially qualified template IDs, (i.e.
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:519]:
```suggestion
  Previously JSON API had supported partially qualified template IDs, such as
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 17]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  simply ``<module>:<entity>``) as an interactive convenience which fails if
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:520]:
```suggestion
  ``<module>:<entity>``, as an interactive convenience, which fails if
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 18]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  there is more than one package with matching template names. **Since this
+  format was not supported for production use and will not work with smart
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:522]:
```suggestion
  format is not supported for production use and does not work with smart
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 19]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
 
 Where:
 
-- ``templateIds`` -- either an array of contract template identifiers or an array containing a single interface identifier to search through. Mixing of template ID's and interface ID's, or specifying more than one interface ID is not allowed.
+- ``templateIds`` -- either an array of contract template identifiers in the
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:1073]:
```suggestion
- ``templateIds``: either an array of contract template identifiers in the
```

File: docs/3.1/docs/json-api/index.rst [hunk 20]
```diff
@@ -506,7 +506,21 @@ HTTP Request
 
 Where:
 
-- ``templateId`` is the contract template identifier, which is formatted as ``"<package ID>:<module>:<entity>"``. As a convenience for interactive API exploration (such as with curl and similar tools), you can also omit the package ID (i.e. specifying the ``templateId`` as ``"<module>:<entity>"``) **if there is only one template with that name across all loaded packages**. Code should always specify the package ID, since it's common to have more versions of a template sharing the same module and entity name but with different package IDs. If the package identifier is not specified and the template cannot be uniquely identified without it, the HTTP JSON API service will report that the specified template cannot be found. **Omitting the package ID is not supported for production use.**
+- ``templateId`` is the contract template identifier which has the following
+  two formats: ``#<package name>:<module>:<entity>`` or ``<package
+  ID>:<module>:<entity>``. 
+  
+  The package name format works with the smart contract upgrading feature so
```
Comment (cathyjung-da) [inline docs/3.1/docs/json-api/index.rst:513]:
```suggestion
  The package name format supports the smart contract upgrading feature,
```

File: docs/3.1/docs/json-api/index.rst [hunk 21]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  that contracts with different package IDs but the same package name can be
```
Comment (cathyjung-da) [inline docs/3.1/docs/json-api/index.rst:514]:
```suggestion
  allowing contracts with different package IDs but the same package name to be
```

File: docs/3.1/docs/json-api/index.rst [hunk 22]
```diff
@@ -506,7 +506,21 @@ HTTP Request
...
+  used in a uniform manner.
```
Comment (cathyjung-da) [inline docs/3.1/docs/json-api/index.rst:515]:
```suggestion
  used uniformly.
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 23]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
...
+  same format as in the :ref:`create request <create-request>`.  It can also
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:1074]:
```suggestion
  same format as in the :ref:`create request <create-request>` or
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 24]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
...
+  be an array containing a single interface identifier to search through.
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:1075]:
```suggestion
  an array containing a single interface identifier to search through.
```

File: docs/3.1/docs/json-api/index.rst [hunk 25]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
 
 Where:
 
-- ``templateIds`` -- either an array of contract template identifiers or an array containing a single interface identifier to search through. Mixing of template ID's and interface ID's, or specifying more than one interface ID is not allowed.
+- ``templateIds`` -- either an array of contract template identifiers in the
```
Comment (cathyjung-da) [inline docs/3.1/docs/json-api/index.rst:1073]:
```suggestion
- ``templateIds``: either an array of contract template identifiers in the
```

File: docs/3.1/docs/json-api/index.rst [hunk 26]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
...
+  same format as in the :ref:`create request <create-request>`.  It can also
```
Comment (cathyjung-da) [inline docs/3.1/docs/json-api/index.rst:1074]:
```suggestion
  same format as in the :ref:`create request <create-request>` or
```

File: docs/3.1/docs/json-api/index.rst [hunk 27]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
...
+  be an array containing a single interface identifier to search through.
```
Comment (cathyjung-da) [inline docs/3.1/docs/json-api/index.rst:1075]:
```suggestion
  an array containing a single interface identifier to search through.
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 28]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
...
+  Mixing of template ID's and interface ID's, or specifying more than one
+  interface ID is not allowed.
 - ``query`` -- search criteria to apply to the specified ``templateIds``, formatted according to the :doc:`search-query-language`.
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:1078]:
```suggestion
- ``query``: search criteria to apply to the specified ``templateIds``, formatted according to the :doc:`search-query-language`.
```

File: docs/2.9.0/docs/json-api/index.rst [hunk 29]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
...
 - ``readers`` -- *optional* non-empty list of parties to query as; must be a subset of the actAs/readAs parties in the JWT
```
Comment (cathyjung-da) [inline docs/2.9.0/docs/json-api/index.rst:1079]:
```suggestion
- ``readers``: *optional* non-empty list of parties to query as; must be a subset of the actAs and readAs parties in the JWT.
```

File: docs/3.1/docs/json-api/index.rst [hunk 30]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
...
+  Mixing of template ID's and interface ID's, or specifying more than one
+  interface ID is not allowed.
 - ``query`` -- search criteria to apply to the specified ``templateIds``, formatted according to the :doc:`search-query-language`.
```
Comment (cathyjung-da) [inline docs/3.1/docs/json-api/index.rst:1078]:
```suggestion
- ``query``: search criteria to apply to the specified ``templateIds``, formatted according to the :doc:`search-query-language`.
```

File: docs/3.1/docs/json-api/index.rst [hunk 31]
```diff
@@ -1056,7 +1070,11 @@ HTTP Request
...
 - ``readers`` -- *optional* non-empty list of parties to query as; must be a subset of the actAs/readAs parties in the JWT
```
Comment (cathyjung-da) [inline docs/3.1/docs/json-api/index.rst:1079]:
```suggestion
- ``readers``: *optional* non-empty list of parties to query as; must be a subset of the actAs and readAs parties in the JWT.
```

**Review by cathyjung-da — APPROVED** [review]
Only made some minor suggestions. Otherwise LGTM.

**Review by raphael-speyer-da — APPROVED** [review]

**Pull request merged**